samples/apps/bookinfo/cluster_extension.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+#
+# Copyright 2017 Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Script to run on a VM with VPC connectivity to the cluster, to prepare the DB and details app.
+
+
+# Initialize mysql for the rating demo
+function istioMysql() {
+  # The ratings script is using root/password by default.
+
+  cat <<EOF | sudo mysql
+GRANT ALL PRIVILEGES on *.* to 'root'@'localhost' IDENTIFIED BY 'password';
+FLUSH PRIVILEGES;
+CREATE DATABASE test;
+EOF
+
+  # Create the tables
+  mysql -u root -ppassword < mysql/mysqldb-init.sql
+}
+
+function istioUpdateRatings() {
+  local RATING=${1-2}
+
+  cat <<EOF | mysql -u root -ppassword
+  USE test;
+  update ratings set rating=$RATING where reviewid=1;
```

Comment:
cute. 
though you can put test as mysql arg and avoid the use test and you can also put the update as a -e all in 1 line:
```
mysql -u root -ppassword test -e "update ratings set rating=$RATING where reviewid=1;
```
(and having the user and password hardcoded is a bit scary)

Reply:
I know you can pass a single command - but I think the other style is more readable and can be used for more generic setups. 

Nothing I can do at this point about the hardcoded password. I believe mongodb sample is also wide open and relies only on firewall. If auth is enabled - mTLS will take care of it.